### PR TITLE
[IOTDB-335] Merge different aggregations on same timeseries

### PR DESCRIPTION
**Pull request by samperson1997**
Separate query executions of the same timeseries with different aggregate functions can be optimized

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java [hunk 1]
```diff
@@ -121,23 +127,21 @@ private AggregateResult aggregateOneSeries(int i, Filter timeFilter, QueryContex
         //cal by pageheader
         if (seriesReader.canUseCurrentPageStatistics()) {
           Statistics pageStatistic = seriesReader.currentPageStatistics();
-          aggregateResult.updateResultFromStatistics(pageStatistic);
-          if (aggregateResult.isCalculatedAggregationResult()) {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java]: If one series relates to a   min_time and first_value, we'd better return result if all aggregateResult  has calculated aggregation results.

File: server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java [hunk 2]
```diff
@@ -80,39 +82,43 @@ public QueryDataSet executeWithoutValueFilter(QueryContext context)
...
-  }
 
+    return constructDataSet(Arrays.asList(aggregateResultList));
+  }
 
-  /**
-   * get aggregation result for one series
-   */
-  private AggregateResult aggregateOneSeries(int i, Filter timeFilter, QueryContext context)
+  private List<AggregateResult> aggregateOneSeries(Map.Entry<Path, List<Integer>> series,
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java]: add javadoc and an example

given : 
paths: s1, s2, s3, s1
aggregations: count, sum, count, sum

seriesMap:  
s1 -> 0, 3
s2 -> 2
s3 -> 3

Comment (samperson1997) [reply server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java]: Hi, thanks! I have intended put the example to `mergeSameSeries` method

Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java]: this function could be renamed to  groupAggregationsBySeries

File: server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java (hunk 1 above)
Comment (samperson1997) [reply server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java]: Thanks for your proposal. I use an `aggregateResultList` to store whether the `aggregateResult` is calculated... which is consistent with the `aggregateResultList`. Do you think it is okay?

File: server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java (hunk 2 above)
Comment (samperson1997) [reply server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java]: Done.

File: server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java [hunk 3]
```diff
@@ -129,10 +129,15 @@ public QueryDataSet executeWithoutValueFilter(QueryContext context)
...
-          aggregateResult.updateResultFromStatistics(chunkStatistics);
-          if (aggregateResult.isCalculatedAggregationResult()) {
-            isCalculatedList.set(i, true);
+          if (Boolean.FALSE.equals(isCalculatedList.get(i))) {
+            AggregateResult aggregateResult = aggregateResultList.get(i);
+            aggregateResult.updateResultFromStatistics(chunkStatistics);
+            if (aggregateResult.isCalculatedAggregationResult()) {
+              isCalculatedList.set(i, true);
+            }
+            if (isCalculatedList.stream().allMatch(element -> element == Boolean.TRUE)) {
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/query/executor/AggregationExecutor.java]: replace this with an counter

**Pull request merged**